src/cerbos/sdk/_async/_hub/_store.py

```diff
@@ -0,0 +1,408 @@
+    Some of the error subclasses contain additional details about the error such as validation failures and ignored files.
+    - AbortedError: call cancelled or timed out
+    - AuthenticationFailedError: invalid credentials
+    - CannotModifyGitConnectedStoreError: attempting to modify a store connected to Git
+    - ConditionUnsatisfiedError: the condition attached to the call wasn't satisfied
+    - InvalidRequestError: the arguments passed to the call are invalid
+    - NoUsableFilesError: none of the provided files can be stored
+    - OperationDiscardedError: the operation does not modify store state
+    - PermissionDeniedError: permission denied for the action
+    - TooManyFailuresError: the client has had too many failed requests and should cool down
+    - UnknownError: catch-all for errors that are not specifically handled
+    - ValidationFailureError: the request was rejected due to validation failures
+    """
+
+    _store_stub: store_pb2_grpc.CerbosStoreServiceStub
+
+    def __init__(
+        self,
+        credentials: Optional[Credentials] = None,
+        api_endpoint: Optional[str] = None,
+        timeout_secs: Optional[float] = None,
+    ):
+        super(AsyncCerbosHubStoreClient, self).__init__(
+            credentials, api_endpoint, timeout_secs
+        )
+        self._store_stub = store_pb2_grpc.CerbosStoreServiceStub(self._channel)
+
+    @handle_store_errors
+    @_CIRCUIT_BREAKER
+    async def replace_files(
+        self,
+        store_id: str,
+        message: str,
+        contents: Union[bytes, Iterable[File]],
+        version_must_equal: Optional[int] = None,
+        change_details: Optional[ChangeDetails] = None,
+    ) -> ReplaceFilesResponse:
+        """
+        Overwrite the store such that it only contains the files included in this request.
+        Raises OperationDiscardedError if the store is already at the desired state.
+        """
+
+        if not (store_id and store_id.strip()):
+            raise InvalidRequestError(ValueError("store_id is required"))
+
+        _change_details: Optional[store_pb2.ChangeDetails] = None
+        if change_details is None:
+            _change_details = store_pb2.ChangeDetails(
+                description=message,
+                uploader=store_pb2.ChangeDetails.Uploader(name="cerbos-sdk-python"),
+            )
+        else:
+            _change_details = change_details.raw
+
+        _condition: Optional[store_pb2.ReplaceFilesRequest.Condition] = None
+        if version_must_equal:
+            _condition = store_pb2.ReplaceFilesRequest.Condition(
+                store_version_must_equal=version_must_equal
+            )
+
+        req = store_pb2.ReplaceFilesRequest(
+            store_id=store_id, condition=_condition, change_details=_change_details
+        )
+        req: store_pb2.ReplaceFilesRequest
+        if isinstance(contents, bytes):
+            if len(contents) < _MIN_ZIP_SIZE or len(contents) > _MAX_ZIP_SIZE:
```

Comment:
I think python lets you inline these double bounded conditionals just FYI

```suggestion
            if not (_MIN_ZIP_SIZE <= len(contents) <= _MAX_ZIP_SIZE):
```